File: week-3/minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
    if (!points || points.length === 0) return 0;
    
    points = points.sort((a,b) => a[0]-b[0]);
    let count = 0, min_end = Infinity;
    
    for (let point of points) {
        if (point[0] > min_end) {
            ++count;
            min_end = point[1];
        } else
            min_end = Math.min(min_end, point[1]);
    }
    
    return ++count;
};

File: week-3/minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if not points or len(points) == 0: return 0
        
        points.sort(key=lambda x: x[0])
        min_end = float('inf')
        count = 0

        for point in points:
            if point[0] > min_end:
                min_end = point[1]
                count += 1
            else:
                min_end = min(min_end, point[1])
            
        return count+1
